src/requests/request.ts

```diff
@@ -196,46 +218,54 @@ async function handleResponseNotOk(
   response: Response,
   url: string,
 ): Promise<void> {
-  let message = "";
+  let originalMessage = "";
   let errorDetails;
+  let isUnsupportedModelError = false;
+  const newSdkUrl = "https://github.com/googleapis/js-genai";
+  // Patterns for newer models likely requiring the new SDK
+  const unsupportedModelPatterns = [/gemini-2/, /gemini-pro-2.0/, /gemini-ultra-2.0/, /gemini-flash-2.5/, /gemini-pro-2.5/, /gemini-2.5-pro/, /gemini-2.5-pro-exp/, /gemini-2.0-flash/];
+
+  // Extract model name from the URL
+  let requestedModel = "unknown";
+  // Match /v.../models/model-name: or /v.../tunedModels/model-name:
+  const modelNameMatch = url.match(/\/(v\d+(?:beta)?\d*)\/(?:models|tunedModels)\/([^:]+):/);
+  if (modelNameMatch && modelNameMatch[2]) {
+      requestedModel = modelNameMatch[2];
+  }
+
   try {
     const json = await response.json();
-    message = json.error.message;
-    if (json.error.details) {
-      message += ` ${JSON.stringify(json.error.details)}`;
-      errorDetails = json.error.details;
+    originalMessage = json.error?.message || ""; // Safely access message
+    errorDetails = json.error?.details; // Safely access details
+
+    // Check 1: Explicit model name check
+    if (unsupportedModelPatterns.some(pattern => pattern.test(requestedModel))) {
+        // If the requested model *is* one of the new ones, assume this error is due to SDK incompatibility.
+        isUnsupportedModelError = true;
+    }
+
+    // Append original details to message if they exist and aren't already in the message
+    if (errorDetails && !originalMessage.includes(JSON.stringify(errorDetails))) {
+      originalMessage += ` ${JSON.stringify(errorDetails)}`;
     }
+
   } catch (e) {
-    // ignored
+    // Error parsing JSON is ignored, proceed with original status/text
+  }
+
+  let finalMessage = `Error fetching from ${url.toString()}: [${response.status} ${response.statusText}] ${originalMessage || 'Server responded with an error.'}`; // Provide default if message is empty
+
+  if (isUnsupportedModelError) {
```

Comment:
this might be a breaking change if user has client side logic to process the `error message`. Maybe a safer approach is to log a warning message to the console?